src/lib.rs

```diff
@@ -76,7 +75,7 @@ mod tests {
         assert_eq!(name, repo_name);
 
         // Add repo to group
-        loaded_group.add_repo(repo).await.expect("Unable to add repo to group");
+        loaded_group.add_repo(repo.clone()).await.expect("Unable to add repo to group");
```

Comment:
I think instead of adding a repo groups should be able to load a repo from a DHT key + encryption key or create a new repo

Reply:
I believe what you are describing is covered by `get_repo`